File: front-end/src/app/components/proposition-responsable/appel-offreDto.ts
import { Proposition } from "./proposition-responsable";

export class AppelOffreDto {
    id?: string;
    dateDebut: Date;
    dateFin: Date;
    propositions: Proposition[];
   
  
    constructor(dateDebut: Date, dateFin: Date) {
      this.dateDebut = dateDebut;
      this.dateFin = dateFin;
      this.propositions = [];
      
    }
   


}

File: Back-end/src/main/java/com/gestresmat/springjwt/DTO/AppelOffreDto.java
package com.gestresmat.springjwt.DTO;

import java.util.Date;
import java.util.List;

public class AppelOffreDto {
    private Long id;
    private Date dateDebut;
    private Date dateFin;

    private List<PropositionDto> propositions;

    public Long getId() {
        return id;
    }

    public void setId(Long id) {
        this.id = id;
    }

    public Date getDateDebut() {
        return dateDebut;
    }

    public void setDateDebut(Date dateDebut) {
        this.dateDebut = dateDebut;
    }

    public Date getDateFin() {
        return dateFin;
    }

    public void setDateFin(Date dateFin) {
        this.dateFin = dateFin;
    }

    public List<PropositionDto> getPropositions() {
        return propositions;
    }

    public void setPropositions(List<PropositionDto> propositions) {
        this.propositions = propositions;
    }
}
